### PR TITLE
Using ParameterSet in SingleFreeTrackParameters

### PR DESCRIPTION
**Pull request by FabianKlimpel**
This PR replaces the plain storage of data by the `ParameterSet` as performed in `SingleTrackParameters`. 
It is a component of #165.

### CODE REVIEW

File: Core/include/Acts/EventData/SingleFreeTrackParameters.hpp [hunk 1]
```diff
@@ -89,10 +87,10 @@ class SingleFreeTrackParameters {
   /// be retrieved
   ///
   /// @return Value of the requested parameter uncertainty
-  template <unsigned int par,
+  template <FreeParametersIndices par,
```
Comment (msmk0) [inline Core/include/Acts/EventData/SingleFreeTrackParameters.hpp:90]: Naming: to follow our naming convention `par` should be something like `kIndex` or `kIdx`.

File: Core/include/Acts/EventData/SingleFreeTrackParameters.hpp [hunk 2]
```diff
@@ -169,10 +167,10 @@ class SingleFreeTrackParameters {
   ///
   /// @note The context is not used here but makes the API consistent with
   /// @c SingleCurvilinearTrackParameters and @c SingleBoundTrackParameters
-  template <unsigned int par,
+  template <FreeParametersIndices par,
```
Comment (msmk0) [inline Core/include/Acts/EventData/SingleFreeTrackParameters.hpp:170]: Naming: as above. `par` should be `kIndex` or `kIdx`.

File: Core/include/Acts/EventData/SingleFreeTrackParameters.hpp (hunk 1 above)
Comment (FabianKlimpel) [reply Core/include/Acts/EventData/SingleFreeTrackParameters.hpp]: Done

File: Core/include/Acts/EventData/SingleFreeTrackParameters.hpp (hunk 2 above)
Comment (FabianKlimpel) [reply Core/include/Acts/EventData/SingleFreeTrackParameters.hpp:170]: done

**Pull request merged**